### PR TITLE
docs: Update search in UG

### LINKED ISSUE #158: [PE-D][Tester E] Search Feature does not find intended medicine
Started a completely new instance of meditracker. Added a medicine with name medicine1, and tried to search for medicine with name medicine1, however, it shows no medications found

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: itsmejr257/ped#6

### LINKED ISSUE #160: [PE-D][Tester C] Search function does not return required medication
Search function is supposed to return the medicine that is called. When calling search -n MedC, MedC is supposed to be returned. However, no search results are found although medC exists in the list.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: cayennegoh/ped#2

### LINKED ISSUE #180: [PE-D][Tester C] Search function does not return items added
Added a item which causes side effect "causescrazy". 
When calling both -a causescrazy and -s causescrazy, the medicine name, MedB should appear. However, no search results are found.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: cayennegoh/ped#8

### LINKED ISSUE #193: [PE-D][Tester A] Unable to search name of medications
Unable to find the medications with the same name in the program.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: Haotian199/ped#8

### LINKED ISSUE #196: [PE-D][Tester A] Search of side effect of medication lists medications that were not added
The medications listed above are not added into the list. However, when searching for medications that has the side effects of dizziness, these medications are listed. Unclear whether the listed medications that produce dizziness is supposed to be the added medications or all the medications in general.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: Haotian199/ped#9

### LINKED ISSUE #197: [PE-D][Tester D] Unexpected output from search
The following commands give me unexpected search output (medicines not in my storage).
```
meditracker> search -n m
Here are the search results:
1. Paracetamol; Treats: Fever, Pain; May cause: Nausea, Liver damage (in high doses)
2. Diazepam; Treats: Anxiety, Muscle spasms; May cause: Drowsiness, Confusion
3. Amoxicillin; Treats: Bacterial infections; May cause: Nausea, Diarrhea, Rash
4. Metformin; Treats: Type 2 diabetes; May cause: Nausea, Diarrhea
5. Alprazolam; Treats: Anxiety; May cause: Drowsiness, Memory problems
6. Omeprazole; Treats: Acid reflux; May cause: Headache, Nausea
7. Amlodipine; Treats: High blood pressure; May cause: Swelling, Dizziness
8. Simvastatin; Treats: High cholesterol; May cause: Muscle pain, Liver damage (rare)
9. Metoprolol; Treats: High blood pressure; May cause: Dizziness, Fatigue
10. Morphine; Treats: Pain; May cause: Constipation, Drowsiness
```

```
search -n l
Here are the search results:
1. Paracetamol; Treats: Fever, Pain; May cause: Nausea, Liver damage (in high doses)
2. Loratadine; Treats: Allergy symptoms; May cause: Headache, Fatigue
3. Amoxicillin; Treats: Bacterial infections; May cause: Nausea, Diarrhea, Rash
4. Ciprofloxacin; Treats: Bacterial infections; May cause: Nausea, Diarrhea, Tendon rupture (rare)
5. Insulin; Treats: Diabetes; May cause: Hypoglycemia, Injection site reactions
6. Losartan; Treats: High blood pressure; May cause: Dizziness, Fatigue
7. Levothyroxine; Treats: Hypothyroidism; May cause: Hair loss, Weight changes
8. Alprazolam; Treats: Anxiety; May cause: Drowsiness, Memory problems
9. Omeprazole; Treats: Acid reflux; May cause: Headache, Nausea
10. Lisinopril; Treats: High blood pressure; May cause: Dizziness, Cough
11. Amlodipine; Treats: High blood pressure; May cause: Swelling, Dizziness
12. Hydrochlorothiazide; Treats: High blood pressure; May cause: Dizziness, Low potassium
13. Metoprolol; Treats: High blood pressure; May cause: Dizziness, Fatigue
```

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: fxe025/ped#9

### PR DESCRIPTION
**Pull request by kyuichyi**
# Checklist for non-implementation related PR
Closes #158, closes #160, closes #180, closes #193, closes #196, closes #197 
- [ ] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [ ] Developer Guide contributions (if any)
  - [x] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [ ] Miscellaneous and team-based admin tasks (if any)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -649,21 +648,24 @@ SUCCESS: Medicine has been deleted
...
 ```
 Usage:
 	search [-n name] [-i illness] [-s sideEffects] [-a allFields] [-h]
 Options:
-	-n name                 Name of medication
-	-i illness              Illness that the medication is used for
-	-s sideEffects          Side effects of the medication
-	-a allFields            Search all fields
-	-h                      Prints this help message
+
```
Comment (e0958902) [inline docs/UserGuide.md:663]: You could remove this additional space to follow the format we have.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -649,21 +648,24 @@ SUCCESS: Medicine has been deleted
 
 ## Search
 
-To search for a medication in the library, you can type 'search', followed by the field you want to search for and the keyword 
-`keyword`.
+MediTracker has a built-in medication library where you can search for common medications. This library contains 
+information on the medication name, the illness that the medication is used for, and the side effects of the medication.
+
```
Comment (e0958902) [inline docs/UserGuide.md:653]: Perhaps you can add a Note admonition box here to somehow alert confused users that this search command is searching in another file, and not in the current medication list in 'list -t all'

**Review by e0958902 — CHANGES_REQUESTED** [review]
Refer to the requested changes.
Afterwards, I would recommend you to generate this branch User Guide webpage(and attach the link) for us to view the rendered markdown in HTML. 
The rest looks good to me.

File: docs/UserGuide.md (hunk 1 above)
Comment (kyuichyi) [reply docs/UserGuide.md]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/238/commits/0dc2d4a8f35532244568b60b7666f4e8aa6308ce

**Review by e0958902 — APPROVED** [review]
All looks good to me. Approving and merging immediately.

**Pull request merged**